src/framework/piral-base/src/api.ts

```diff
@@ -2,10 +2,10 @@ import type { PiletMetadata, EventEmitter, PiletApi, PiletApiExtender } from './
 
 export function initializeApi(target: PiletMetadata, events: EventEmitter): PiletApi {
   return {
-    on: events.on,
-    once: events.once,
-    off: events.off,
-    emit: events.emit,
+    on: events.on.bind(events),
```

Comment:
Alternatively, solve this already in `events.ts`

Reply:
Makes sense, will do!

Reply:
Done. This really is a much better fix for the issue.